pkg/manager/manager.go

```diff
@@ -45,6 +49,11 @@ func New() *CollectorManager {
 	manager.PrometheusCollector.ContainersMetrics = &manager.MetricCollector.ContainersMetrics
 	manager.PrometheusCollector.ProcessMetrics = &manager.MetricCollector.ProcessMetrics
 	manager.PrometheusCollector.SamplePeriodSec = SamplePeriodSec
+	// configure the wather
+	manager.Watcher = kubernetes.NewObjListWatcher()
+	manager.Watcher.Mx = &manager.PrometheusCollector.Mx
```

Comment:
create Watcher with Mx instead of overwrite.

Reply:
It does not overwrite. It uses the shared locker across other packages.